astroplan/core.py

```diff
@@ -162,56 +164,336 @@ def altaz(self, time, target=None, obswl=None):
+                    format='jd')
+
+    def _calc_riseset(self, time, target, prev_next, rise_set, horizon, N=150):
+        '''
+        Time at next rise/set of `target`.
 
         Parameters
         ----------
         time : `~astropy.time.Time`
-        """
-        raise NotImplementedError()
+            Time of observation
 
-    def midnight(self, time):
-        """
-        Returns the local, solar midnight time.
+        target : `~astropy.coordinates.SkyCoord`
+            Position of target or multiple positions of that target
+            at multiple times (if target moves, like the Sun)
+
+        prev_next : str - either 'previous' or 'next'
+            Test next rise/set or previous rise/set
+
+        rise_set : str - either 'rising' or 'setting'
+            Compute prev/next rise or prev/next set
+
+        location : `~astropy.coordinates.EarthLocation`
+            Location of observer
+
+        horizon : `~astropy.units.Quantity`
+            Degrees above/below actual horizon to use
+            for calculating rise/set times (i.e.,
+            -6 deg horizon = civil twilight, etc.)
+
+        N : int
+            Number of altitudes to compute when searching for
+            rise or set.
+
+        Returns
+        -------
+        ret1 : `~astropy.time.Time`
+            Time of rise/set
+        '''
+        if prev_next == 'next':
+            times = time + np.linspace(0, 1, N)*u.day
+        else:
+            times = time + np.linspace(-1, 0, N)*u.day
+
+        altitudes = self.altaz(times, target).alt
+        horizon_crossing_limits = self._horiz_cross(times, altitudes, rise_set,
+                                                    horizon,
+                                                    return_limits=True,
+                                                    return_alts=True)
+        return self._two_point_interp(*horizon_crossing_limits, horizon=horizon)
+
+    @u.quantity_input(horizon=u.deg)
+    def calc_rise(self, time, target, which='nearest', horizon=0*u.degree):
+        '''
+        Calculate rise time.
+
+        Compute time of the next/previous/nearest rise of the `target`
+        object, where "rise" is defined as the time when the `target`
+        transitions from altitudes below the `horizon` to above the
+        `horizon`.
 
         Parameters
         ----------
         time : `~astropy.time.Time`
-        """
-        raise NotImplementedError()
+            Time of observation
 
-    def sunset(self, time, **kwargs):
-        """
-        Returns the local sunset time.
+        target : coordinate object (i.e. `~astropy.coordinates.SkyCoord`, `~astroplan.FixedTarget`)
+            Target celestial object
 
-        The default sunset returned is the next one to occur.
+        which : str - can be "next", "previous", or "nearest"; defaults to "nearest"
+            Choose which sunrise relative to the present `time` would you
```

Comment:
Here time needs to be in double-backticks ... you should always use double-backticks for variable names (and single-backticks for types.
If you check the HTML docs, you'll see that otherwise a link to here will be generated:
https://docs.python.org/2/library/time.html#module-time